### ISSUE TITLE
How to create a dataset for Movinet transfer learning?

### ISSUE THREAD

**Issue by aafaqin** [tensorflow/models#10455]
official/vision/beta/projects/movinet/movinet_tutorial.ipynb

How to create your own dataset for doing transfer learning on movinet. I was able to see the tutorial for ucf101 dataset. But the dataset format is not clearly visible it automatically downloads and creates  'train_dataset'  I could see some TFrecord files getting downloaded in a Ucf101 dataset. but how to train your own videos?

**Comment by Hyperparticle** [comment]
If you want to try something fast and easy you can try to [convert your videos to numpy arrays](https://stackoverflow.com/questions/42163058/how-to-turn-a-video-into-numpy-array), preprocess them, and then wrap the arrays in a [TF dataset](https://www.tensorflow.org/api_docs/python/tf/data/Dataset). For a large number of examples it's best to [write a custom dataset](https://www.tensorflow.org/datasets/add_dataset) and use our [video dataloader](https://github.com/tensorflow/models/blob/de44470ffa22d588f3fc4ce6c1f4a789ca636eab/official/vision/beta/dataloaders/video_input.py) to load video tensors and preprocess them automatically.

Does this help?

**Comment by tucachmo2202** [comment]
> If you want to try something fast and easy you can try to [convert your videos to numpy arrays](https://stackoverflow.com/questions/42163058/how-to-turn-a-video-into-numpy-array), preprocess them, and then wrap the arrays in a [TF dataset](https://www.tensorflow.org/api_docs/python/tf/data/Dataset). For a large number of examples it's best to [write a custom dataset](https://www.tensorflow.org/datasets/add_dataset) and use our [video dataloader](https://github.com/tensorflow/models/blob/de44470ffa22d588f3fc4ce6c1f4a789ca636eab/official/vision/beta/dataloaders/video_input.py) to load video tensors and preprocess them automatically.
> 
> Does this help?

Hi, could you provide some sample data or more detail tutorial for transfer learning?

**Comment by Hyperparticle** [comment]
I made some recent changes to the [movinet_tutorial.ipynb](https://github.com/tensorflow/models/blob/master/official/projects/movinet/movinet_tutorial.ipynb) that should work with fine-tuning UCF-101 on a single GPU for 3 epochs to reach 85% top-1 validation accuracy. The transfer learning for other datasets should be nearly the same, it would just need some set up in the tf.data.Dataset. You can create and load your own TFRecords using [this tutorial](https://www.tensorflow.org/tutorials/load_data/tfrecord).

These are fairly straightforward if you are familiar with setting up a tf.data.Dataset with image data, but I understand if this is not familiar to you I can try to update the tutorial with these steps. Are there any particular parts of this setup that you need more help on?

**Comment by aeD0T** [comment]
@Hyperparticle  I was hoping you provide more details on setting up the tutorial using custom videos. Please and Thank you

**Comment by RolandasRazma** [comment]
@Hyperparticle thanks for example on fine tuning, works like a charm for base model. What changes needed for streaming one? I get error about input shape mismatch if I setup model same way as in "Export a Streaming Model to TensorFlow Lite for Mobile"
